packages/pgsearch/client.js

```diff
@@ -129,6 +204,14 @@ module.exports = class PgClient {
     let refs = await context.references();
     let realms = await context.realms();
 
+    if (touched) {
+      touched[`${type}/${id}`] = touchCounter++;
```

Comment:
This looks like a bug. This isn't updating a shared counter, it's updating a local variable that never goes back out of this method.

More generally, this looks like an opportunity to make the touch-tracking and invalidation part of pgsearch's API. Like:

```
let batch = client.beginBatch();
// saveDocument and deleteDocument would only be available on `batch`, not directly on `client` anymore
await batch.saveDocument({ context });
await batch.deleteDocument(...whatever);

// this would automatically run invalidations. All the `touched` tracking happens inside the `batch`.
await batch.done();
```

This also has the benefit of giving us a logical place to add database transactions in the future if we want that (such that only an entire batch becomes visible at once, never the intermediate states during indexing).

Reply:
nice. will do